src/main/java/com/onehouse/metadata_extractor/TableMetadataUploaderService.java

```diff
@@ -104,10 +104,21 @@ private CompletableFuture<Boolean> uploadInstantsInTableBatch(List<Table> tables
         .thenComposeAsync(
             getTableMetricsCheckpointResponse -> {
               if (getTableMetricsCheckpointResponse.isFailure()) {
-                log.error(
-                    "Error encountered when fetching checkpoint, skipping table processing. status code: {} message {}",
-                    getTableMetricsCheckpointResponse.getStatusCode(),
-                    getTableMetricsCheckpointResponse.getCause());
+                int statusCode = getTableMetricsCheckpointResponse.getStatusCode();
+                StringBuilder errorMessageBuilder = new StringBuilder();
+                errorMessageBuilder.append(
+                    "Error encountered when fetching checkpoint, skipping table processing. ");
```

Comment:
Instead of adding this here, can we add it to the cause at the source: https://github.com/onehouseinc/hudi-metadata-extractor/blob/467bbfb91f8ad00fb51f0e613e8ab2673ddc5e8e/src/main/java/com/onehouse/api/OnehouseApiClient.java#L158

the cause field gets printed everywhere

Reply:
Sounds good. Addressed your feedback.